extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -35,17 +37,27 @@ export interface DotNetCommandOptions {
 export class NetCoreTool {
 
 	private static _outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel(projectsOutputChannel);
+	private osPlatform: string = os.platform();
+	private netCoreSdkInstalledVersion: string | undefined;
+	private netCoreError: netCoreFindError = netCoreFindError.noError;
 
 	public async findOrInstallNetCore(): Promise<boolean> {
-		if (!this.isNetCoreInstallationPresent && vscode.workspace.getConfiguration(DBProjectConfigurationKey)[NetCoreDoNotAskAgainKey] !== true) {
+		this.netCoreError = netCoreFindError.noError;
```

Comment:
There's a potential for a race condition here if something in the background tries to run some command while we're still searching for it. Would probably be better to just set this after the if check when we know that it's been found.

Reply:
Makes sense. Updated it.